go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go

```diff
@@ -292,6 +283,60 @@ func (manager *LogicalSwitchManager) AllocateNextIPs(nodeName string) ([]*net.IP
 	return ipnets, nil
 }
 
+func (manager *LogicalSwitchManager) AllocateHybridOverlay(nodeName string, hybridOverlayAnnotation []string) ([]*net.IPNet, error) {
+	if len(hybridOverlayAnnotation) > 0 {
+		var allocateAddresses []*net.IPNet
+		for _, ip := range hybridOverlayAnnotation {
+			allocateAddresses = append(allocateAddresses, &net.IPNet{IP: net.ParseIP(ip).To4(), Mask: net.CIDRMask(32, 32)})
+		}
+		err := manager.AllocateIPs(nodeName, allocateAddresses)
+		if err != nil {
+			return nil, err
+		}
+		return allocateAddresses, nil
+	}
+	// if we are not provided with any addresses
+	manager.RLock()
+	defer manager.RUnlock()
+
+	lsi, ok := manager.cache[nodeName]
+	if !ok {
+		return nil, fmt.Errorf("node %s not found in the logical switch manager cache", nodeName)
+	}
+	// determine ipams are ipv4
+	var ipv4IPAMS []ipam.Interface
+	for _, ipam := range lsi.ipams {
+		if utilnet.IsIPv4(ipam.CIDR().IP) {
+			ipv4IPAMS = append(ipv4IPAMS, ipam)
+		}
+	}
+	var allocatedAddresses []*net.IPNet
+	for _, ipv4IPAM := range ipv4IPAMS {
+		hostSubnet := ipv4IPAM.CIDR()
+		potentialHybridIFAddress := util.GetNodeHybridOverlayIfAddr(&hostSubnet)
+		err := ipv4IPAM.Allocate(potentialHybridIFAddress.IP)
+		if err == ipam.ErrAllocated {
+			// allocate NextIP
+			allocatedipv4, err := ipv4IPAM.AllocateNext()
+			if err != nil {
+				_ = manager.ReleaseIPs(nodeName, allocatedAddresses)
+				return nil, fmt.Errorf("cannot allocate hybrid overlay interface address for node/subnet %s/%s (%+v)", nodeName, hostSubnet, err)
+
+			}
+			if err == nil {
```

Comment:
since we will only get here if err == nil, this line (326) and line 329 are not needed, right?

Reply:
there is an error that is declared in this block on line 320 that is what this error check on 326 is for

Reply:
ack!